app_integrations/config.py

```diff
@@ -47,9 +47,14 @@ class States(object):
         SUCCEEDED = 'succeeded'
         FAILED = 'failed'
 
-    def __init__(self, config, **kwargs):
+    class Events(object):
+        """Events object to encapsulate various acceptable events"""
+        SUCCESSIVE_INVOKE = 'successive_invoke'
+
+    def __init__(self, config, event=None, **kwargs):
         dict.__init__(self, config, **kwargs)
         self._validate_config()
+        self._event = event or {}
```

Comment:
why not just make the default `{}`?

Reply:
Good thought, but: 
![image](https://user-images.githubusercontent.com/13773789/31917729-f369d618-b80d-11e7-92a6-5714b1b9c926.png)

Reply:
why is that dangerous?

Reply:
See here:
http://pylint-messages.wikidot.com/messages:w0102

> To a Python novice, this may seem like a reasonable way to default to an empty array for the extras param. What really happens is that this "default" array gets created as a persistent object, and every invocation of my_method that doesn't specify an extras param will be using that same list object—any changes to it will persist and be carried to every other invocation!

Reply:
oh right, I forgot Python did that